src/pudl/workspace/setup.py

```diff
@@ -1,15 +1,114 @@
 """Tools for setting up and managing PUDL workspaces."""
 import importlib
+import os
 import pathlib
 import shutil
+from pathlib import Path
+from typing import IO
 
 import yaml
+from dotenv import load_dotenv
 
 import pudl.logging_helpers
 
 logger = pudl.logging_helpers.get_logger(__name__)
 
 
+def get_defaults(
+    input_dir: str | None = None,
+    output_dir: str | None = None,
+    yaml_file: IO | None = None,
+    default_pudl_yaml: Path | None = Path.home() / ".pudl.yml",
+) -> dict[str, str]:
+    """Derive PUDL workspace paths from specified input/output directories.
+
+    Determines input/output directory locations from YAML, then overrides with
+    env vars, then overrides with keywords passed in.
```

Comment:
So if no kwargs are specified, it uses the env var values. If no env vars a specified it uses the .pudl.yaml file. If there isn't a .pudl.yaml file it raises an error?

Reply:
It's looking for 2 values to pass in to `derive_paths` and knows to find them in those three locations. It throws the error if it can't find anything.

Reply:
That's the idea. And the only time we really would expect to be specifying kwargs is during testing to point to temp dirs

Reply:
Got it. I have my PUDL_OUTPUT set but when I call pudl.workspace.setup.get_defaults() I get paths that use the path stored in .pudl.yml. Is this expected?

Reply:
Can you paste what your `PUDL_OUTPUT` and `PUDL_INPUT` are set to, what's in .pudl.yml, and and also what `get_defaults` actually returns?

Reply:
.pudl.yml:
```
pudl_in: /Users/bendnorman/catalyst/pudl-work
pudl_out: /Users/bendnorman/catalyst/pudl-work
```

`PUDL_OUTPUT`:
```
/Users/bendnorman/catalyst/dagster-pudl-work/pudl_output/
```

`PUDL_INPUT` is unset

`get_defualts()`:
```
{'pudl_in': '/Users/bendnorman/catalyst/pudl-work',
 'data_dir': '/Users/bendnorman/catalyst/pudl-work/data',
 'settings_dir': '/Users/bendnorman/catalyst/pudl-work/settings',
 'pudl_out': '/Users/bendnorman/catalyst/pudl-work/output',
 'sqlite_dir': '/Users/bendnorman/catalyst/pudl-work/output',
 'parquet_dir': '/Users/bendnorman/catalyst/pudl-work/output',
 'PUDL_OUTPUT': '/Users/bendnorman/catalyst/pudl-work/output',
 'PUDL_INPUT': '/Users/bendnorman/catalyst/pudl-work/data',
 'ferc1_db': 'sqlite:////Users/bendnorman/catalyst/pudl-work/output/ferc1.sqlite',
 'ferc1_xbrl_db': 'sqlite:////Users/bendnorman/catalyst/pudl-work/output/ferc1_xbrl.sqlite',
 'ferc1_xbrl_datapackage': PosixPath('/Users/bendnorman/catalyst/pudl-work/output/ferc1_xbrl_datapackage.json'),
 'ferc1_xbrl_taxonomy_metadata': PosixPath('/Users/bendnorman/catalyst/pudl-work/output/ferc1_xbrl_taxonomy_metadata.json'),
 'ferc2_xbrl_db': 'sqlite:////Users/bendnorman/catalyst/pudl-work/output/ferc2_xbrl.sqlite',
 'ferc2_xbrl_datapackage': PosixPath('/Users/bendnorman/catalyst/pudl-work/output/ferc2_xbrl_datapackage.json'),
 'ferc2_xbrl_taxonomy_metadata': PosixPath('/Users/bendnorman/catalyst/pudl-work/output/ferc2_xbrl_taxonomy_metadata.json'),
 'ferc6_xbrl_db': 'sqlite:////Users/bendnorman/catalyst/pudl-work/output/ferc6_xbrl.sqlite',
 'ferc6_xbrl_datapackage': PosixPath('/Users/bendnorman/catalyst/pudl-work/output/ferc6_xbrl_datapackage.json'),
 'ferc6_xbrl_taxonomy_metadata': PosixPath('/Users/bendnorman/catalyst/pudl-work/output/ferc6_xbrl_taxonomy_metadata.json'),
 'ferc60_xbrl_db': 'sqlite:////Users/bendnorman/catalyst/pudl-work/output/ferc60_xbrl.sqlite',
 'ferc60_xbrl_datapackage': PosixPath('/Users/bendnorman/catalyst/pudl-work/output/ferc60_xbrl_datapackage.json'),
 'ferc60_xbrl_taxonomy_metadata': PosixPath('/Users/bendnorman/catalyst/pudl-work/output/ferc60_xbrl_taxonomy_metadata.json'),
 'ferc714_xbrl_db': 'sqlite:////Users/bendnorman/catalyst/pudl-work/output/ferc714_xbrl.sqlite',
 'ferc714_xbrl_datapackage': PosixPath('/Users/bendnorman/catalyst/pudl-work/output/ferc714_xbrl_datapackage.json'),
 'ferc714_xbrl_taxonomy_metadata': PosixPath('/Users/bendnorman/catalyst/pudl-work/output/ferc714_xbrl_taxonomy_metadata.json'),
 'pudl_db': 'sqlite:////Users/bendnorman/catalyst/pudl-work/output/pudl.sqlite',
 'censusdp1tract_db': 'sqlite:////Users/bendnorman/catalyst/pudl-work/output/censusdp1tract.sqlite'}
```

Reply:
False alarm... I was working on a branch that didn't have these changes yet :/

Reply:
(closure: we hopped on a call and resolved this - the behavior was coming from some old code that didn't have these changes merged in yet)